SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverGameplayDebuggerContext.cpp

```diff
@@ -242,6 +255,84 @@ void USpatialNetDriverGameplayDebuggerContext::TickServer()
 
 		It.RemoveCurrent();
 	}
+
+	for (auto& Pair : TrackedEntities)
+	{
+		if (Pair.Value.Component.TrackPlayer)
+		{
+			if (!NetDriver->HasServerAuthority(Pair.Key))
+			{
+				// Do not track player if we don't have authority
+				continue;
+			}
+
+			TWeakObjectPtr<AGameplayDebuggerCategoryReplicator> ReplicatorWeakObjectPtr = Pair.Value.ReplicatorWeakObjectPtr;
+			AGameplayDebuggerCategoryReplicator* Replicator = ReplicatorWeakObjectPtr.Get();
+			if (Replicator == nullptr)
+			{
+				// Not unexpected - assume latency and wait for actor to appear
+				continue;
+			}
+
+			const AActor* PlayerControllerActor = Cast<AActor>(Replicator->GetReplicationOwner());
+			if (!ensureMsgf(PlayerControllerActor != nullptr, TEXT("Serious error: Cannot find player controller for replicated gameplay debugger actor")))
+			{
+				continue;
+			}
+
+			check(NetDriver->LoadBalanceStrategy);
+			auto GetActorVirtualWorkerId = [](const AActor& InActor, const USpatialNetDriver& InNetDriver, const SpatialGDK::FSubView& InSubView)->VirtualWorkerId
+			{
+				check(InNetDriver.PackageMap);
+
+				const Worker_EntityId EntityId = InNetDriver.PackageMap->GetEntityIdFromObject(&InActor);
+				if (!ensureMsgf(EntityId != SpatialConstants::INVALID_ENTITY_ID, TEXT("Serious error: Cannot find actor in package map")))
+				{
+					return SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
+				}
+
+				for (const SpatialGDK::ComponentData& Data : InSubView.GetView()[EntityId].Components)
```

Comment:
Yeah I could see how this might become expensive if we were tracking lots of entities. Is that the case though? I assumed the debugger could only track one actor at once, but TrackedEntities suggests otherwise.

Reply:
It will only ever track one actor per player, but we still need to be able to track multiple entities because there could be multiple players using the debugger at the same time.

Reply:
Multiple players using the debugger independently does not work at the moment, they share the same debugger even in native. Therefore, I could have stored a single entity for now, however I used a vector for future expansion - if this then brings performance problems, we should/could optimize then.